ambry-store/src/main/java/com.github.ambry.store/StorageManager.java

```diff
@@ -376,6 +396,67 @@ int getCompactionThreadCount() {
     return count;
   }
 
+  /**
+   * This method is called by Offline-To-Dropped transition. Any errors/exceptions will be thrown and converted to
+   * {@link StateTransitionException}. The error/exception is also recorded in certain metric for alerting purpose.
+   */
+  private void resumeDecommission(ReplicaId replica) throws Exception {
```

Comment:
Can we structure the code so that we don't copy the logic from replication manager and stats manager here?
We can probably having a reference of ReplicationManagerListener and StatsManagerListener here and just call
listener.becomeInactiveFromStandby and listener.becomeOfflineFromInactive.

Reply:
Not quite sure if I am able to make it but let me double check the feasibility.

Reply:
I made it!  The code is now more concise :)